src/NServiceBus.Transport.SQS/Extensions/TransportMessageExtensions.cs

```diff
@@ -43,33 +41,27 @@ static class TransportMessageExtensions
 
             var s3GetResponse = await s3Client.GetObjectAsync(getObjectRequest, cancellationToken)
                 .ConfigureAwait(false);
-            int contentLength = (int)s3GetResponse.ContentLength;
-            var buffer = arrayPool.Rent(contentLength);
-            using (var memoryStream = new MemoryStream(buffer))
+
+            using (var memoryStream = new MemoryStream())
             {
-                await s3GetResponse.ResponseStream.CopyToAsync(memoryStream, 81920, cancellationToken).ConfigureAwait(false);
-                return Tuple.Create((ReadOnlyMemory<byte>)buffer.AsMemory(0, contentLength), buffer);
+                await s3GetResponse.ResponseStream.CopyToAsync(memoryStream).ConfigureAwait(false);
+                return memoryStream.ToArray();
             }
         }
 
-        static Tuple<ReadOnlyMemory<byte>, byte[]> ConvertBody(string body, ArrayPool<byte> arrayPool)
+        static byte[] ConvertBody(string body)
         {
             var encoding = Encoding.UTF8;
-            // TODO check if we need fallback
             try
             {
-                return Tuple.Create(new ReadOnlyMemory<byte>(Convert.FromBase64String(body)), default(byte[]));
+                return Convert.FromBase64String(body);
             }
             catch (FormatException)
             {
-                var length = encoding.GetMaxByteCount(body.Length);
-                var buffer = arrayPool.Rent(length);
-                var writtenBytes = encoding.GetBytes(body, 0, body.Length, buffer, 0);
-                return Tuple.Create((ReadOnlyMemory<byte>)buffer.AsMemory(0, writtenBytes), buffer);
+                return encoding.GetBytes(body);
             }
         }
 
-        static readonly Tuple<ReadOnlyMemory<byte>, byte[]>
-            EmptyMessage = Tuple.Create(new ReadOnlyMemory<byte>(), default(byte[]));
+        static readonly byte[] EmptyMessage = new byte[0];
```

Comment:
Array.Empty would work too. If you want to keep the static field due to expressiveness you could assign Array.Empty instead of new byte[0]